tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorTests.cs

```diff
@@ -147,6 +144,48 @@ public void Any_with_a_nested_Any()
                 "{G: {$elemMatch: {D: \"Don't\", \"S\": {$elemMatch: {E: null, D: \"Delilah\" }}}}}");
         }
 
+        [Fact]
+        public void Any_with_a_not_and_a_predicate_with_not_contains()
+        {
+            var x = new[] { 1, 2 };
+
+            Assert(
+                c => !c.M.Any(a => !x.Contains(a)),
+                "{ M : { '$not' : { '$elemMatch' : { '$nin' : [1, 2] } } } }");
+        }
+
+        [Fact]
+        public void Any_with_a_not_and_a_predicate_with_contains()
+        {
+            var x = new[] { 1, 2 };
+
+            Assert(
+                c => !c.M.Any(a => x.Contains(a)),
+                "{ M : { '$not' : { '$elemMatch' : { '$in' : [1, 2] } } } }");
```

Comment:
I'm concerned that we've transformed the expression so much and that we may have inadvertently altered the semantics of the query. Why not translate more directly to:

    { $nor : [ M : { $elemMatch : { $in : [1, 2] } } ] }

Reply:
The same thing as in previous answer